pandas/core/reshape/reshape.py

```diff
@@ -1039,7 +1039,8 @@ def get_empty_frame(data) -> DataFrame:
         return out
 
     else:
-        dummy_mat = np.eye(number_of_cols, dtype=dtype).take(codes, axis=0)
+        # take on axis=1 + transpose to ensure ndarray layout is column-major
+        dummy_mat = np.eye(number_of_cols, dtype=dtype).take(codes, axis=1).T
```

Comment:
@jorisvandenbossche investigating a perf regression in `reshape.GetDummies.time_get_dummies_1d` current best guess is this line.  Would doing `arr.T.take(codes, axis=0)` be equivalent here?  That's about 7x faster for me locally.

Reply:
@jorisvandenbossche gentle ping; time_get_dummies_1d is the biggest perf regression im seeing vs 1.2.0